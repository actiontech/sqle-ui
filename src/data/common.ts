export enum ResponseCode {
  SUCCESS = 0,
}

export enum SystemRole {
  admin = 'admin',
}

export enum ModalSize {
  big = 1000,
  mid = 800,
}

export const RuleListDefaultTabKey = 'ALL';
export const instanceListDefaultKey = 'all';
export const ruleTemplateListDefaultKey = 'all';

export const ModalFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 7 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 13 },
  },
};

export const PageFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 8 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 12 },
    md: { span: 10 },
  },
};

export const PageBigFormLayout = {
  labelCol: {
    xs: { span: 24 },
    sm: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 14 },
    // md: { span: 1 },
  },
};

export const FilterFormLayout = {
  labelCol: { span: 7 },
  wrapperCol: { span: 17 },
};

export const FilterFormColLayout = {
  xs: 24,
  sm: 12,
  xl: 8,
  xxl: 6,
};

export const filterFormButtonLayoutFactory = (
  smOffset = 0,
  xlOffset = 0,
  xxlOffset = 0
) => ({
  xs: 24,
  sm: {
    span: 12,
    offset: smOffset,
  },
  xl: {
    span: 8,
    offset: xlOffset,
  },
  xxl: {
    span: 6,
    offset: xxlOffset,
  },
});

export const FilterFormRowLayout = {
  gutter: 24,
};

export enum LoginTypeEnum{
  'ldap'='ldap',
  'sqle'='sqle',
}