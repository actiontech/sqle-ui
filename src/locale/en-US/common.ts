// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: 'username',
  password: 'password',

  unknownError: 'Unknown Error',

  ok: 'Ok',
  cancel: 'Cancel',
  submit: 'Submit',
  close: 'Close',
  edit: 'Edit',
  delete: 'Delete',
  reset: 'Reset',
  search: 'Search',
  retry: 'Retry',
  back: 'Back',
  upload: 'Upload',
  resetAndClose: 'Reset Form And Close',
  operateSuccess: 'Operate successfully',
  prevStep: 'Prev',
  nextStep: 'Next',

  expansion: 'Expand',
  collapse: 'Collapse',

  theme: {
    light: 'Light Theme',
    dark: 'Dark Theme',
  },

  logout: 'Logout',
  account: 'Account',

  nav: {
    title: 'SQLe',
  },

  request: {
    noticeFailTitle: 'Request Failed',
  },

  form: {
    placeholder: {
      input: 'Please input {{name}}',
      select: 'Please select {{name}}',
      searchInput: 'Input the {{name}} to search',
      searchSelect: 'Select the {{name}} to search',
    },
    rule: {
      require: 'Please input {{name}}',
      passwordNotMatch: 'The two passwords that you entered do not match!',
      email: 'The input is not valid E-mail!',
    },
  },
};
