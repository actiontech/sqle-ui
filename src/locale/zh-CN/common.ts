// eslint-disable-next-line import/no-anonymous-default-export
export default {
  username: '用户名',
  password: '密码',

  unknownError: '未知错误...',
  unknownStatus: '未知状态...',

  ok: '确认',
  cancel: '取消',
  submit: '提交',
  close: '关闭',
  edit: '编辑',
  modify: '修改',
  delete: '删除',
  reset: '重置',
  resetAll: '重置所有内容',
  search: '搜索',
  retry: '重试',
  back: '返回',
  more: '更多',
  upload: '选择文件上传',
  resetAndClose: '重置表单并关闭',
  operateSuccess: '操作成功',
  operate: '操作',

  prevStep: '上一步',
  nextStep: '下一步',

  expansion: '展开',
  collapse: '收起',

  showAll: '查看所有',
  showDetail: '查看详情',

  in: '在',
  on: '的',
  and: '且',
  at: '中',
  preview: '预览',

  theme: {
    light: '明亮模式',
    dark: '暗黑模式',
  },

  logout: '退出登录',
  account: '个人中心',

  nav: {
    title: 'SQLE',
  },

  request: {
    noticeFailTitle: '请求错误',
  },

  time: {
    hour: '小时',
    year: '年',
    month: '月',
    day: '天',
    no: '日',
    week: '星期',
    minute: '分钟',
    per: '每',
  },

  form: {
    placeholder: {
      input: '请输入{{name}}',
      select: '请选择{{name}}',
      searchInput: '请输入要搜索的 {{name}}',
      searchSelect: '请选择要搜索的 {{name}}',
    },
    rule: {
      require: '必须填写{{name}}',
      selectFile: '必须选择一个文件',
      passwordNotMatch: '两次密码输入不一致',
      email: '请输入有效的邮箱地址',
      startWithLetter: '必须要以字母开头',
      onlyLetterAndNumber: '只能包含字母、数字、中划线和下划线',
    },
  },

  cron: {
    mode: {
      select: '可视化选择',
      manual: '手工填写',
    },
    errorMessage: {
      invalid: '无效的cron表达式',
      mustBeString: 'cron表达式必须是一个字符串',
      mustBeArray: '变更的值必须是一个数组',
      lenMustBeFive: 'cron表达式必须只包含（分钟 小时 日期 月份 星期）5个元素',
      onlyHaveValidChar: 'cron表达式只能包含数字中划线(-),斜线(/),和逗号(,)',
      limit:
        '您的表达式中包含不合法的数值范围， minute(0-59), hour(0-23), day(1,31), month(1-12), week(0-6)',
    },
  },
};
