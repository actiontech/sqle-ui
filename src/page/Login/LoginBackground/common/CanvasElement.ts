import { IPosition } from './index.type';

abstract class CanvasElement {
  abstract position: IPosition;
  abstract canvas: HTMLCanvasElement;
  abstract context: CanvasRenderingContext2D;
  abstract draw(): void;
  abstract update(reachPosition: IPosition): void;
}

export default CanvasElement;
