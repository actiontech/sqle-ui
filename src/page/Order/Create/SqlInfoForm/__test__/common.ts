import instance from '../../../../../api/instance';
import { resolveThreeSecond } from '../../../../../testUtils/mockRequest';

export const mockGetInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceV2');
  spy.mockImplementation(() =>
    resolveThreeSecond({
      instance_name: 'db1',
      db_host: '20.20.20.2',
      db_port: '3306',
      db_user: 'root',
      db_type: 'mysql',
      desc: '',
      workflow_template_name: 'workflow-template-name-1',
      rule_template: {
        name: 'not_submit_test_rule33',
        is_global_rule_template: true,
      },
    })
  );

  return spy;
};
