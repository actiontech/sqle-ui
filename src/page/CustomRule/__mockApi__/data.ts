import { ICustomRuleResV1 } from '../../../api/common';
import { CustomRuleResV1LevelEnum } from '../../../api/common.enum';

export const customRules: ICustomRuleResV1[] = [
  {
    db_type: 'mysql',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.notice,
    rule_id: '1',
    desc: 'name1',
    rule_script: 'w+',
    type: 'rule_type1',
  },
  {
    db_type: 'oracle',
    annotation: 'desc desc desc',
    level: CustomRuleResV1LevelEnum.error,
    rule_id: '2',
    desc: 'name2',
    rule_script: 'w+',
    type: 'rule_type2',
  },
];
