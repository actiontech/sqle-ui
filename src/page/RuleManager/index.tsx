import { Card, PageHeader, Tabs, TabsProps } from 'antd';
import { useTranslation } from 'react-i18next';
import GlobalRuleTemplate from '../GlobalRuleTemplate';
import CustomRule from '../CustomRule';
import useNavigate from '../../hooks/useNavigate';
import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';
import { t } from '../../locale';
import { SQLE_BASE_URL } from '../../data/common';

const tabItems: TabsProps['items'] = [
  {
    label: t('ruleTemplate.globalRuleTemplateListTitle'),
    children: <GlobalRuleTemplate />,
    key: `${SQLE_BASE_URL}rule/template`,
  },
  /* IFTRUE_isEE */
  {
    label: t('customRule.title'),
    children: <CustomRule />,
    key: `${SQLE_BASE_URL}rule/custom`,
  },
  /* FITRUE_isEE */
];

const RuleManager: React.FC = () => {
  const { t } = useTranslation();
  const navigate = useNavigate();
  const location = useLocation();

  const [activeKey, setActiveKey] = useState(tabItems[0].key);

  useEffect(() => {
    const currentKey = tabItems.find((v) =>
      location.pathname.startsWith(v.key)
    )?.key;
    if (currentKey) {
      setActiveKey(currentKey);
    }
  }, [location.pathname]);

  return (
    <article className="rule-template-page-namespace">
      <PageHeader title={t('ruleManager.pageTitle')} ghost={false}>
        {t('ruleManager.pageDesc')}
      </PageHeader>

      <section className="padding-content">
        <Card>
          <Tabs
            activeKey={activeKey}
            onChange={(key) => {
              navigate(key);
            }}
            items={tabItems}
          />
        </Card>
      </section>
    </article>
  );
};

export default RuleManager;
