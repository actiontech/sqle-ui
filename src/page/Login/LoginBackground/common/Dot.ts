import { random } from '../../../../utils/Math';
import CanvasElement from './CanvasElement';
import { IPosition } from './index.type';

class Dot extends CanvasElement {
  public canvas: HTMLCanvasElement;
  public context: CanvasRenderingContext2D;
  public position: IPosition = { x: 0, y: 0 };
  private size: number;
  private color: string;
  private slope!: number;
  private offset!: number;
  private speed: number;
  private randomNum: number;
  private autoMove: boolean = true;

  constructor(
    canvas: HTMLCanvasElement,
    speed?: number,
    size?: number,
    color?: string,
    slope?: number,
    offset?: number
  ) {
    super();
    this.canvas = canvas;
    this.context = canvas.getContext('2d') as CanvasRenderingContext2D;
    this.size = size ?? 3;
    this.color = color ?? '#fff';
    this.speed = speed ?? 1;
    this.randomNum = Math.random() * 5 + 5;
    this.initPosition();
    if (!slope || !offset) {
      this.randomTrajectory();
    }
  }

  public draw(): void {
    this.context.beginPath();
    this.context.fillStyle = this.color;
    this.context.arc(
      this.position.x,
      this.position.y,
      this.size,
      0,
      Math.PI * 2
    );
    this.context.fill();
    this.context.closePath();
  }

  public update(reachPosition?: IPosition): void {
    let { x, y } = this.position;
    if (!reachPosition) {
      if (!this.autoMove) {
        return;
      }
      x += this.speed;
      if (x <= 0 || x >= this.canvas.width) {
        x = 0;
      }
      y = this.slope * x + this.offset;
    } else {
      x = x + ((reachPosition.x - x) * this.randomNum) / 50;
      y = y + ((reachPosition.y - y) * this.randomNum) / 50;
    }
    this.position = { x, y };
  }

  public closeAutoMove() {
    this.autoMove = false;
  }

  public openAutoMove() {
    this.randomTrajectory();
    this.autoMove = true;
  }

  public randomTrajectory() {
    const slopeBig = random(1, this.canvas.height / 2000);
    const slopeSmall = Math.random();
    const direction = Math.random() > 0.5 ? 1 : -1;
    const slope = (Math.random() > 0.5 ? slopeBig : slopeSmall) * direction;
    const offset = this.position.y - slope * this.position.x;
    this.slope = slope;
    this.offset = offset;
  }

  public initPosition() {
    this.position = {
      x: random(1, this.canvas.width - 1),
      y: random(1, this.canvas.height - 1),
    };
  }
}

export default Dot;
