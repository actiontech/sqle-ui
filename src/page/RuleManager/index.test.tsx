/* eslint-disable no-console */
import { cleanup, screen } from '@testing-library/react';
import RuleManager from '.';
import {
  renderLocationDisplay,
  renderWithMemoryRouter,
} from '../../testUtils/customRender';
import { MemoryRouter } from 'react-router-dom';
import { SQLE_BASE_URL } from '../../data/common';
import { fireEvent } from '@testing-library/react';

describe('test rule manager', () => {
  const warn = console.warn;
  beforeAll(() => {
    console.warn = jest.fn();
    (console.warn as any).mockImplementation((message: any) => {
      if (message.includes('No routes matched location')) {
        return;
      }
      warn(message);
    });
  });
  afterAll(() => {
    console.warn = warn;
  });

  beforeEach(() => {
    jest.useFakeTimers();
  });
  afterEach(() => {
    jest.useRealTimers();
  });
  test('should match snapshot', () => {
    const { container, rerender } = renderWithMemoryRouter(
      <RuleManager />,
      undefined,
      {
        initialEntries: [`${SQLE_BASE_URL}rule/template`],
      }
    );

    expect(container).toMatchSnapshot();

    expect(
      screen.getByText('ruleTemplate.globalRuleTemplateListTitle').parentNode
    ).toHaveClass('ant-tabs-tab-active');

    rerender(
      <MemoryRouter initialEntries={[`${SQLE_BASE_URL}rule/template/create`]}>
        <RuleManager />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
    expect(
      screen.getByText('ruleTemplate.globalRuleTemplateListTitle').parentNode
    ).toHaveClass('ant-tabs-tab-active');

    cleanup();
    const { container: container1 } = renderWithMemoryRouter(
      <RuleManager />,
      undefined,
      {
        initialEntries: [`/`],
      }
    );
    expect(container1).toMatchSnapshot();
  });

  test('should change location when click tabs tab btn', async () => {
    const [, LocationComponent] = renderLocationDisplay();
    renderWithMemoryRouter(
      <>
        <RuleManager />
        <LocationComponent />
      </>,
      undefined,
      {
        initialEntries: [`${SQLE_BASE_URL}rule/custom`],
      }
    );

    expect(screen.getByText('customRule.title').parentNode).toHaveClass(
      'ant-tabs-tab-active'
    );

    fireEvent.click(
      screen.getByText('ruleTemplate.globalRuleTemplateListTitle')
    );
    expect(
      screen.getByText('ruleTemplate.globalRuleTemplateListTitle').parentNode
    ).toHaveClass('ant-tabs-tab-active');
    expect(screen.getByTestId('location-display')).toHaveTextContent(
      `${SQLE_BASE_URL}rule/template`
    );
  });
});
