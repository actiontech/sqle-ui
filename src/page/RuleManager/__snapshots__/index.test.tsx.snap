// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test rule manager should match snapshot 1`] = `
<div>
  <article>
    <div
      class="ant-page-header"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ruleManager.pageTitle"
          >
            ruleManager.pageTitle
          </span>
        </div>
      </div>
      <div
        class="ant-page-header-content"
      >
        ruleManager.pageDesc
      </div>
    </div>
    <section
      class="padding-content"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-tabs ant-tabs-top"
          >
            <div
              class="ant-tabs-nav"
              role="tablist"
            >
              <div
                class="ant-tabs-nav-wrap"
              >
                <div
                  class="ant-tabs-nav-list"
                  style="transform: translate(0px, 0px);"
                >
                  <div
                    class="ant-tabs-tab ant-tabs-tab-active"
                    data-node-key="/rule/template"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/template"
                      aria-selected="true"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/template"
                      role="tab"
                      tabindex="0"
                    >
                      ruleTemplate.globalRuleTemplateListTitle
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                    data-node-key="/rule/custom"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/custom"
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/custom"
                      role="tab"
                      tabindex="0"
                    >
                      customRule.title
                    </div>
                  </div>
                  <div
                    class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                  />
                </div>
              </div>
              <div
                class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
              >
                <button
                  aria-controls="rc-tabs-test-more-popup"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-hidden="true"
                  class="ant-tabs-nav-more"
                  id="rc-tabs-test-more"
                  style="visibility: hidden; order: 1;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="ant-tabs-content-holder"
            >
              <div
                class="ant-tabs-content ant-tabs-content-top"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="rc-tabs-test-tab-/rule/template"
                  class="ant-tabs-tabpane ant-tabs-tabpane-active"
                  id="rc-tabs-test-panel-/rule/template"
                  role="tabpanel"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </article>
</div>
`;

exports[`test rule manager should match snapshot 2`] = `
<div>
  <article>
    <div
      class="ant-page-header"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ruleManager.pageTitle"
          >
            ruleManager.pageTitle
          </span>
        </div>
      </div>
      <div
        class="ant-page-header-content"
      >
        ruleManager.pageDesc
      </div>
    </div>
    <section
      class="padding-content"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-tabs ant-tabs-top"
          >
            <div
              class="ant-tabs-nav"
              role="tablist"
            >
              <div
                class="ant-tabs-nav-wrap"
              >
                <div
                  class="ant-tabs-nav-list"
                  style="transform: translate(0px, 0px);"
                >
                  <div
                    class="ant-tabs-tab ant-tabs-tab-active"
                    data-node-key="/rule/template"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/template"
                      aria-selected="true"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/template"
                      role="tab"
                      tabindex="0"
                    >
                      ruleTemplate.globalRuleTemplateListTitle
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                    data-node-key="/rule/custom"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/custom"
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/custom"
                      role="tab"
                      tabindex="0"
                    >
                      customRule.title
                    </div>
                  </div>
                  <div
                    class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                  />
                </div>
              </div>
              <div
                class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
              >
                <button
                  aria-controls="rc-tabs-test-more-popup"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-hidden="true"
                  class="ant-tabs-nav-more"
                  id="rc-tabs-test-more"
                  style="visibility: hidden; order: 1;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="ant-tabs-content-holder"
            >
              <div
                class="ant-tabs-content ant-tabs-content-top"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="rc-tabs-test-tab-/rule/template"
                  class="ant-tabs-tabpane ant-tabs-tabpane-active"
                  id="rc-tabs-test-panel-/rule/template"
                  role="tabpanel"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </article>
</div>
`;

exports[`test rule manager should match snapshot 3`] = `
<div>
  <article>
    <div
      class="ant-page-header"
    >
      <div
        class="ant-page-header-heading"
      >
        <div
          class="ant-page-header-heading-left"
        >
          <span
            class="ant-page-header-heading-title"
            title="ruleManager.pageTitle"
          >
            ruleManager.pageTitle
          </span>
        </div>
      </div>
      <div
        class="ant-page-header-content"
      >
        ruleManager.pageDesc
      </div>
    </div>
    <section
      class="padding-content"
    >
      <div
        class="ant-card ant-card-bordered"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-tabs ant-tabs-top"
          >
            <div
              class="ant-tabs-nav"
              role="tablist"
            >
              <div
                class="ant-tabs-nav-wrap"
              >
                <div
                  class="ant-tabs-nav-list"
                  style="transform: translate(0px, 0px);"
                >
                  <div
                    class="ant-tabs-tab ant-tabs-tab-active"
                    data-node-key="/rule/template"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/template"
                      aria-selected="true"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/template"
                      role="tab"
                      tabindex="0"
                    >
                      ruleTemplate.globalRuleTemplateListTitle
                    </div>
                  </div>
                  <div
                    class="ant-tabs-tab"
                    data-node-key="/rule/custom"
                  >
                    <div
                      aria-controls="rc-tabs-test-panel-/rule/custom"
                      aria-selected="false"
                      class="ant-tabs-tab-btn"
                      id="rc-tabs-test-tab-/rule/custom"
                      role="tab"
                      tabindex="0"
                    >
                      customRule.title
                    </div>
                  </div>
                  <div
                    class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                  />
                </div>
              </div>
              <div
                class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
              >
                <button
                  aria-controls="rc-tabs-test-more-popup"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-hidden="true"
                  class="ant-tabs-nav-more"
                  id="rc-tabs-test-more"
                  style="visibility: hidden; order: 1;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="ellipsis"
                    class="anticon anticon-ellipsis"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="ellipsis"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="ant-tabs-content-holder"
            >
              <div
                class="ant-tabs-content ant-tabs-content-top"
              >
                <div
                  aria-hidden="false"
                  aria-labelledby="rc-tabs-test-tab-/rule/template"
                  class="ant-tabs-tabpane ant-tabs-tabpane-active"
                  id="rc-tabs-test-panel-/rule/template"
                  role="tabpanel"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </article>
</div>
`;
